src/RecoveryLoadingController.js

```diff
@@ -29,6 +29,7 @@ define(['okta', 'util/FormController'], function (Okta, FormController) {
       })
       .fail(function () {
         self.options.appState.trigger('loading', false);
+        self.appState.trigger('removeLoading');
```

Comment:
is this a problem in other controllers? This will fix the recovery flows, but what about other flows?

Reply:
Luckily, other places that remove loading change the page itself on app state change. So we do not encounter this anywhere else. The one place where this is possible is primaryAuthController and it already has this `removeLoading` call in it.